tests/test_system_metrics.py

```diff
@@ -0,0 +1,137 @@
+import time
+
+import pytest
+
+from dvclive import Live
+from dvclive.system_metrics import CPUMetricsCallback, get_cpus_metrics
+from dvclive.utils import parse_metrics
+
+
+def mock_psutil(mocker):
+    mocker.patch(
+        "dvclive.system_metrics.psutil.cpu_percent",
+        return_value=[10, 20, 30, 40, 50, 60],
+    )
+    mocker.patch("dvclive.system_metrics.psutil.cpu_count", return_value=6)
+
+    mocked_virtual_memory = mocker.MagicMock()
+    mocked_virtual_memory.percent = 20
+    mocked_virtual_memory.total = 4 * 1024**3
+
+    mocked_disk_io_counters = mocker.MagicMock()
+    mocked_disk_io_counters.read_bytes = 2 * 1024**2
+    mocked_disk_io_counters.read_time = 1
+    mocked_disk_io_counters.write_bytes = 2 * 1024**2
+    mocked_disk_io_counters.write_time = 1
+
+    mocking_dict = {
+        "virtual_memory": mocked_virtual_memory,
+        "disk_io_counters": mocked_disk_io_counters,
+    }
+    for function_name, return_value in mocking_dict.items():
+        mocker.patch(
+            f"dvclive.system_metrics.psutil.{function_name}",
+            return_value=return_value,
+        )
+
+
+@pytest.mark.parametrize(
+    ("metric_name"),
+    [
+        "system/cpu/usage_avg_percent",
+        "system/cpu/usage_max_percent",
+        "system/cpu/count",
+        "system/cpu/parallelism_percent",
+        "system/cpu/ram_usage_percent",
+        "system/cpu/ram_total_GB",
+        "system/io/read_speed_MB",
+        "system/io/write_speed_MB",
+    ],
+)
+def test_get_cpus_metrics(mocker, metric_name):
+    mock_psutil(mocker)
+    metrics = get_cpus_metrics()
+    assert metric_name in metrics
+
+
+@pytest.mark.parametrize(
+    ("duration", "interval", "plot"),
+    [
+        (1, 0.5, True),
+        (2.0, 1, True),
+    ],
+)
+def test_cpumetricscallback_with_plot(tmp_dir, duration, interval, plot):
+    with Live(
+        tmp_dir,
+        save_dvc_exp=False,
+        callbacks=[CPUMetricsCallback(duration, interval, plot)],
+    ) as live:
+        time.sleep(duration * 2)
```

Comment:
can we do w/o sleep?

also, I'm afraid that logic where we just wait for thread with a sleep can be unreliable - there is not real guarantee for it to exit? it can lead to flakey tests

Reply:
I wanted to test the CPUMetrics' behavior and how it gathers metrics over time. How do you think we should do that?

Reply:
@iterative/dvc Any ideas on how to test this?

Reply:
The problem is that a step could be very long in real experiments (like one hour) and you probably want to have a finer resolution for hardware metrics :)

Reply:
Should `Live` itself run a loop to re-collect from callbacks time to time?